### PR TITLE
thrift_proxy: use per-upstream sequence ids

### PR DESCRIPTION
**Pull request by zuercher**
Rather than use the downstream connection's sequence ID, track
sequence IDs per upstream connection.

*Risk Level*: low
*Testing*: unit tests
*Docs Changes*: n/a
*Release Notes*: n/a

Signed-off-by: Stephan Zuercher <stephan@turbinelabs.io>

### CODE REVIEW

**Comment by zuercher** [comment]
@brirams @rgs1 @fishcakez if you'd like to have a look

File: source/extensions/filters/network/thrift_proxy/conn_state.h [hunk 1]
```diff
@@ -12,6 +12,11 @@ namespace ThriftProxy {
  */
 class ThriftConnectionState : public Tcp::ConnectionPool::ConnectionState {
 public:
+  /**
+   * @return int32_t the next Thrift sequence id to use for this connection.
+   */
+  int32_t nextSequenceId() { return next_sequence_id_++; }
```
Comment (rgs1) [inline source/extensions/filters/network/thrift_proxy/conn_state.h]: what's the plan for overflows here?

**Comment by zuercher** [comment]
@junr03 do you mind taking an overview pass on this so I can merge it?

File: test/extensions/filters/network/thrift_proxy/conn_state_test.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace ThriftProxy {
+
+// Test behavior of nextSequenceId()
+TEST(ThriftConnectionStateTest, NextSequenceId) {
+  ThriftConnectionState cs;
+
+  EXPECT_EQ(0, cs.nextSequenceId());
+  EXPECT_EQ(1, cs.nextSequenceId());
+
+  cs.setNextSequenceIdForTest(std::numeric_limits<int32_t>::max());
```
Comment (junr03) [inline test/extensions/filters/network/thrift_proxy/conn_state_test.cc]: I guess my suggestion makes this test a bit annoying. Unless we allow to seed the sequence id in the constructor and you seed it at max - 1?

Comment (zuercher) [reply test/extensions/filters/network/thrift_proxy/conn_state_test.cc]: I'll make a constructor to handle this.

**Pull request merged**